File: typescript/c_mockups.ts
// const axios = require("axios");
let api = require('./z_utils')

// API request mockup
// For the following arguments set ("2020-11-01", "EUR", "USD,JPY")
// The response should be
// {"rates":{"JPY":122.36,"USD":1.1698},"base":"EUR","date":"2020-10-30"}

async function getCurrencyRatesFromCustomTs(date:string, base_currency:string, currencies:string[]){
    const URL = "https://api.exchangeratesapi.io/"
    let url = `${URL}${date}?base=${base_currency}&symbols=${currencies}`
    let resp = await api.get(url)
    if (resp.status == 200){
        return resp.data.rates
    }
    return {}
}

File: python/c_mockups.py
import urllib
from z_utils import get_request

URL = "https://api.exchangeratesapi.io/{date}?base={curr}&symbols={symbols}"

# API request mockup
# For the following arguments set ("2020-11-01", "EUR", "USD,JPY")
# The response should be
# {"rates":{"JPY":122.36,"USD":1.1698},"base":"EUR","date":"2020-10-30"}

def get_currency_rate_from_urllib(date, base_currency, currencies):
    rates = urllib.urlopen(
        URL.format(
            date=date, 
            curr=base_currency, 
            symbols=currencies
        )
    ).read()
    return rates

def get_currency_rate_from_custom(date, base_currency, currencies):
    rates = get_request(url=URL.format(date=date, curr=base_currency, symbols=currencies))
    return rates

if __name__ == "__main__":
    print(get_currency_rate("2020-11-01", "EUR", "USD,JPY"))